libsigner/src/session.rs

```diff
@@ -217,7 +218,16 @@ impl SignerSession for StackerDBSession {
         for slot_id in slot_ids.iter() {
             let path = stackerdb_get_chunk_path(self.stackerdb_contract_id.clone(), *slot_id, None);
             let chunk = match self.rpc_request("GET", &path, None, &[]) {
-                Ok(body_bytes) => Some(body_bytes),
+                Ok(body_bytes) => {
+                    // Verify that the chunk is not too large
+                    if body_bytes.len() > u32::MAX as usize {
+                        None
+                    } else if body_bytes.len() as u32 > SIGNERS_STACKERDB_CHUNK_SIZE {
```

Comment:
`SIGNERS_STACKERDB_CHUNK_SIZE` is vastly smaller than `u32::MAX`, so let's just use that for comparison

Reply:
I'll admit that I don't fully follow the "let's just use that for comparison" phrase, i.e. what do you mean by _that_ exactly? I'm not familiar with some of the potential gotchas relating to these casting/comparison functions in Rust, so that may be what I'm missing.

Reply:
Use the smaller of the two values here for checking that `body_bytes.len()` small enough.  I.e., only compare `body_bytes.len()` to `SIGNERS_STACKERDB_CHUNK_SIZE`.  You ought to be able to safely cast `SIGNERS_STACKERDB_CHUNK_SIZE` to `usize` since it's pretty small compared to what `usize::MAX` will be on all supported platforms (which is the maximum size of a memory address IIRC).